src/pages/Todo/Todo.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, useState } from "react";
+import { Container, Typography } from "@mui/material";
+import { v4 as uuidv4 } from "uuid";
+import Header from "./components/Header";
+import TodoList from "./components/TodoList";
+import { Task } from "./types";
+import { useSnackbarAlerts } from "../../hooks/useSnackbarAlerts";
+
+const Todo: FC = () => {
+  const [todoList, setToDoList] = useState<Task[]>([]);
+  const { showSuccessSnackbar } = useSnackbarAlerts();
+
+  const addTask = (taskName: string, deadline: number): void => {
+    const newTask: Task = {
+      id: uuidv4(),
+      taskName,
+      deadline,
+      isComplete: false,
+    };
+    setToDoList([...todoList, newTask]);
+  };
+
+  const deleteTask = (taskId: string): void => {
+    setToDoList(todoList.filter((task) => task.id !== taskId));
+    showSuccessSnackbar("Task Deleted Successfully");
+  };
+
+  const completeTask = (taskId: string): void => {
+    setToDoList(
+      todoList.map((task) =>
+        task.id === taskId ? { ...task, isComplete: !task.isComplete } : task
```

Comment:
When working with maps or data iteration we tend to put it in a constant with a good descriptive name. 

We can keep the function as it is but as i said for big scale we need to see the small details.

```suggestion
       const taskList=      todoList.map((task) =>
      {
      const haveSameId=task.id === taskId
       if( haveSameId )return { ...task, isComplete: !task.isComplete } 
      return task
      }
```